gtoolkit/run.sh

```diff
@@ -0,0 +1,224 @@
+    mv "${SMALLTALK_CI_BUILD}"/*.changes "${SMALLTALK_CI_CHANGES}"
+  fi
+
+  if ! vm_is_user_provided; then
+    gtoolkit::prepare_vm
+  fi
+
+  if ! is_file "${SMALLTALK_CI_IMAGE}"; then
+    print_error_and_exit "Failed to prepare image at '${SMALLTALK_CI_IMAGE}'."
+  fi
+}
+
+
+################################################################################
+# Run a Smalltalk script.
+################################################################################
+gtoolkit::run_script() {
+  local script=$1
+  local vm_flags=""
+  local resolved_vm="${config_vm:-${SMALLTALK_CI_VM}}"
+  local resolved_image="$(resolve_path "${config_image:-${SMALLTALK_CI_IMAGE}}")"
+
+  if ! is_travis_build && ! is_headless; then
+    vm_flags="--no-quit"
+  fi
+
+  run_script "${resolved_vm}" "${resolved_image}" eval ${vm_flags} "${script}"
+}
+
+################################################################################
+# Load project into GToolkit image.
+################################################################################
+gtoolkit::load_project() {
+  gtoolkit::run_script "
+    | smalltalkCI |
+    $(conditional_debug_halt)
+    [ | metacello |
+        metacello := Metacello new
+            baseline: 'SmalltalkCI';
+            repository: 'filetree://$(resolve_path "${SMALLTALK_CI_HOME}/repository")';
+            onUpgrade: [ :ex | ex useIncoming ].
+        (Metacello canUnderstand: #onConflictUseIncoming)
+            ifTrue: [ metacello onConflictUseIncoming ]
+            ifFalse: [ metacello onConflict: [ :ex | ex useIncoming ] ].
+        metacello load ]
+            on: Warning
+            do: [ :w | w resume ].
+    smalltalkCI := Smalltalk at: #SmalltalkCI.
+    smalltalkCI load: '$(resolve_path "${config_ston}")'.
+    (smalltalkCI isHeadless or: [ smalltalkCI promptToProceed ])
+      ifTrue: [ smalltalkCI saveAndQuitImage ]
+  "
+}
+
+################################################################################
+# Run tests for project.
+################################################################################
+gtoolkit::test_project() {
+  gtoolkit::run_script "
+    | smalltalkCI |
+    $(conditional_debug_halt)
+    smalltalkCI := Smalltalk
+        at: #SmalltalkCI
+        ifAbsent: [
+            [ | metacello |
+                metacello := Metacello new
+                    baseline: 'SmalltalkCI';
+                    repository: 'filetree://$(resolve_path "${SMALLTALK_CI_HOME}/repository")';
+                    onUpgrade: [ :ex | ex useIncoming ].
+                (Metacello canUnderstand: #onConflictUseIncoming)
+                    ifTrue: [ metacello onConflictUseIncoming ]
+                    ifFalse: [ metacello onConflict: [ :ex | ex useIncoming ] ].
+                metacello load ]
+                    on: Warning
+                    do: [ :w | w resume ].
+            Smalltalk at: #SmalltalkCI ].
+    smalltalkCI test: '$(resolve_path "${config_ston}")'
+  "
+}
```

Comment:
Are these identical to what's used in Pharo?

Reply:
Yes. Ultimately there probably should be an enhancement in GT to run examples as well as tests, but this is beyond my expertise at the moment and Tudor hasn't weighed in yet. It's been a long time (since moving to GT over a year ago) since I've been able to even have an automated build, so I'm eager to get *something* up and running here